doc/compiler/il/GlRegDeps.md

```diff
@@ -0,0 +1,152 @@
+<!--
+Copyright (c) 2021, 2021 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Global Register Dependency
+
+`GlRegDeps` is an IL level construct that models code generator register dependencies. It associates virtual registers with real registers, or pairs virtual registers with real registers.
+
+Register dependencies enforce constraints. They represent that a particular value (IL node) has to be in a particular real register at a particular point in the program. These particular points in the program are usually entry or exit corresponding to basic blocks or extended basic blocks where there is control flow. Fundamentally we can think of global register dependencies as a way of conveying information within real (hardware) registers across block boundaries.
```

Comment:
The word "usually," leaves the reader wondering what other cases might exist.  I think calls are another place where `GlRegDeps` can be used.  Is that correct?  Are there others?

Reply:
My understanding is that those control flow opcodes listed above also require `GlRegDeps` for the branch target if required.

Reply:
I do not believe calls can use GlRegDeps. The "usually" phrase may simply denote that those opcodes are some of the common ways to exit a block, but one could also exit by just fall through, i.e. using none of those opcodes. 

One can only enter a block at the BBStart though, i.e. no complications there.

Reply:
The [`ILOpCode::canHaveGlRegDeps` method](https://github.com/eclipse/omr/blob/7c9f32af80283ebe5ef391f0fbf1a780531ddca0/compiler/il/OMRILOps.hpp#L157) indicates calls can have `GlRegDeps`.  Is there some subset of methods for which they are allowed/required, or is that method out of date?

Reply:
It may have been possible with some of the static languages that were being handled by the same codebase a few years ago but I don't really know of a case personally.